packages/mui-material-next/src/Button/Button.tsx

```diff
@@ -0,0 +1,671 @@
+      pointerEvents: 'none', // Disable link interactions
+      cursor: 'default',
+      color: disabledLabelTextColor,
+      background: disabeldContainerColor[ownerState.variant ?? 'text'],
+      boxShadow: 'var(--md-comp-button-disabled-container-elevation, none)', // Should be md.sys.elevation.level0
+      ...(ownerState.variant === 'outlined' && {
+        border: `1px solid ${
+          theme.vars
+            ? `rgba(${theme.vars.sys.color.onSurfaceChannel} / 0.12)`
+            : alpha(theme.sys.color.onSurface, 0.12)
+        }`,
+      }),
+    },
+  };
+});
+
+const ButtonStartIcon = styled('span', {
+  name: 'MuiButton',
+  slot: 'StartIcon',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [styles.startIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
+  },
+})<{ ownerState: ButtonOwnerState }>(({ ownerState }) => ({
+  display: 'inherit',
+  marginRight: 8,
+  marginLeft: -8,
+  ...(ownerState.size === 'small' && {
+    marginLeft: -2,
+  }),
+  ...commonIconStyles(ownerState),
+}));
+
+const ButtonEndIcon = styled('span', {
+  name: 'MuiButton',
+  slot: 'EndIcon',
+  overridesResolver: (props, styles) => {
+    const { ownerState } = props;
+
+    return [styles.endIcon, styles[`iconSize${capitalize(ownerState.size)}`]];
+  },
+})<{ ownerState: ButtonOwnerState }>(({ ownerState }) => ({
+  display: 'inherit',
+  marginRight: -8,
+  marginLeft: 8,
+  ...(ownerState.size === 'small' && {
+    marginRight: -2,
+  }),
+  ...commonIconStyles(ownerState),
+}));
+
+const Button = React.forwardRef(function Button<
+  BaseComponentType extends React.ElementType = ButtonTypeMap['defaultComponent'],
+>(inProps: ButtonProps<BaseComponentType>, ref: React.ForwardedRef<any>) {
+  const props = useThemeProps({ props: inProps, name: 'MuiButton' });
+  const {
+    action,
+    children,
+    className,
+    color = 'primary',
+    component = 'button',
+    disabled = false,
+    focusableWhenDisabled = false,
+    disableElevation = false,
+    endIcon: endIconProp,
+    focusVisibleClassName,
+    fullWidth = false,
+    LinkComponent = 'a',
+    onBlur,
+    onClick,
+    onContextMenu,
+    onDragLeave,
+    onFocus,
+    onFocusVisible,
+    onKeyDown,
+    onKeyUp,
+    onMouseDown,
+    onMouseLeave,
+    onMouseUp,
+    onTouchEnd,
+    onTouchMove,
+    onTouchStart,
+    size = 'medium',
+    startIcon: startIconProp,
+    tabIndex = 0,
+    type,
+    variant = 'text',
+    ...other
+  } = props;
+
+  const buttonRef = React.useRef<HTMLButtonElement | HTMLAnchorElement | HTMLElement>(null);
+  const handleRef = useForkRef(buttonRef, ref);
+
+  let ComponentProp = component;
+
+  if (ComponentProp === 'button' && (other.href || other.to)) {
+    ComponentProp = LinkComponent;
+  }
+
+  const { focusVisible, setFocusVisible, getRootProps } = useButton({
+    disabled,
+    focusableWhenDisabled,
+    href: props.href,
+    onFocusVisible,
+    tabIndex,
+    // @ts-ignore
```

Comment:
This doesn't feel right. I suppose we could rethink how `useButton` handles link buttons (or even if it should do this at all).

Reply:
I basically replicate what we have in the Material UI button component, as I didn't want to kill of features that are not related to the design system itself. We could come back to it when working on v6, independently of the Material You effort.